app/app.rb

```diff
@@ -0,0 +1,66 @@
+ENV["RACK_ENV"] ||=  "development"
+require 'sinatra/base'
+require "sinatra/flash"
+require_relative 'data_mapper_setup'
+require_relative "models/peep"
+require_relative "models/user"
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  get '/' do
+    erb :index
+  end
+
+  post '/log_in' do
+    user = User.authenticate(params[:username], params[:password])
+   if user
+     session[:user_id] = user.id
+     redirect '/chitter'
+   else
+     flash.now[:errors] = ['Your email or password is incorrect - try sign up!']
+     erb :index
+   end
+ end
+
+  get '/sign_up' do
+    erb :sign_up
+  end
+
+  post '/add_user_data' do
+    user = User.create(username: params[:username], password: params[:password])
+    session[:user_id] = user.id
```

Comment:
Could the lines than set session user_id be refactored to use a helper function? eg:

```
def log_in_user(user_id)
  session[:user_id] = user_id
end
```

That way, the idea of how to log in a user is abstracted into one place.  This makes the mechanism of logging in a user easier to change.  It also makes it clearer what the code is doing.